bedrock/tabzilla/templates/tabzilla/tabzilla.js

```diff
@@ -388,73 +388,81 @@ var Tabzilla = (function (Tabzilla) {
             return false;
         }
 
+        var normalize = function (lang) {
+            return lang.replace(/^(\w+)(?:-(\w+))?$/, function (m, p1, p2) {
+                return p1.toLowerCase() + ((p2) ? '-' + p2.toUpperCase() : '');
+            });
+        };
+
+        // Normalize the every language for easier comparison
+        userLangs = $.map(userLangs, function (lang) { return normalize(lang); });
+        pageLang = normalize(pageLang);
+
+        // If the page language is the user's primary language, there is nothing
+        // to do here
+        if (pageLang === userLangs[0]) {
+            return false;
+        }
+
         var offeredLang;
-        var langLinks = [];
-        var langOptions = [];
+        var availableLangs = {};
+        var $links = $('link[hreflang]');
+        var $options = $('#language option');
+
+        // Make a dictionary from alternate URLs or a language selector. The key
+        // is a language, the value is the relevant <link> or <option> element
+        if ($links.length) {
+            $links.each(function () {
+                availableLangs[normalize(this.hreflang)] = this;
+            });
+        } else if ($options.length) {
+            $options.each(function () {
+                availableLangs[normalize(this.value.match(/^\/?([\w\-]+)/)[1])] = this;
+            });
+        }
```

Comment:
If both `$links.length` and `$options.length` are `0` here, could just `return false;` early and avoid iterating through the user languages below?